### PR TITLE
refactor(metadata): no default graphql operation name

### PR DESCRIPTION
**Pull request by soyuka**

### CODE REVIEW

File: src/GraphQl/Type/FieldsBuilder.php [hunk 1]
```diff
@@ -78,6 +78,7 @@ public function getItemQueryFields(string $resourceClass, Operation $operation,
             return [];
         }
 
+        // TODO: it'd be nice to get rid of this hard coded name
```
Comment (alanpoulain) [inline src/GraphQl/Type/FieldsBuilder.php:81]: Why not using `instanceof` here?

File: src/GraphQl/Type/TypeBuilder.php [hunk 2]
```diff
@@ -104,27 +107,25 @@ public function getResourceObjectType(?string $resourceClass, ResourceMetadataCo
...
                     // Use a new type for the wrapped object only if there is a specific normalization context for the mutation or the subscription.
                     // If not, use the query type in order to ensure the client cache could be used.
                     $useWrappedType = $queryNormalizationContext !== $mutationNormalizationContext;
 
-                    $wrappedOperationName = $operationName;
+                    $wrappedOperation = $operation;
 
                     if (!$useWrappedType) {
-                        $wrappedOperationName = $operation instanceof Query ? $operationName : 'item_query';
+                        $wrappedOperation = $resourceMetadataCollection->getGraphQlOperation(null, forceCollection: $operation instanceof Query);
```
Comment (alanpoulain) [inline src/GraphQl/Type/TypeBuilder.php:126]: Why using a named argument here and not elsewhere?

File: src/GraphQl/Type/TypeConverter.php [hunk 3]
```diff
@@ -82,10 +82,7 @@ public function convertType(Type $type, bool $input, Operation $rootOperation, s
                         } catch (ResourceClassNotFoundException|OperationNotFoundException) {
                         }
                         /** @var Query $enumOperation */
-                        $enumOperation = (new Query())
-                            ->withClass($type->getClassName())
-                            ->withShortName($operation?->getShortName() ?? (new \ReflectionClass($type->getClassName()))->getShortName())
-                            ->withDescription($operation?->getDescription());
+                        $enumOperation = new Query(name: 'item_query', class: $type->getClassName(), shortName: $operation?->getShortName() ?? (new \ReflectionClass($type->getClassName()))->getShortName(), description: $operation?->getDescription());
```
Comment (alanpoulain) [inline src/GraphQl/Type/TypeConverter.php:85]: `name` is not necessary, isn't it?

File: src/GraphQl/Type/TypeBuilder.php (hunk 2 above)
Comment (soyuka) [reply src/GraphQl/Type/TypeBuilder.php:126]: I want to start using them I can revert for the codestyle

File: src/GraphQl/Type/TypeConverter.php (hunk 3 above)
Comment (soyuka) [reply src/GraphQl/Type/TypeConverter.php:85]: we do need a name

File: src/Metadata/GraphQl/Query.php [hunk 4]
```diff
@@ -115,7 +115,7 @@ class: $class,
             fetchPartial: $fetchPartial,
             forceEager: $forceEager,
             priority: $priority,
-            name: $name ?: 'item_query',
+            name: $name,
```
[the remark this answers is not in the document]
Comment (soyuka) [reply src/Metadata/GraphQl/Query.php:118]: because we want to remove logic from our operations

File: src/GraphQl/Type/TypeBuilder.php (hunk 2 above)
Comment (alanpoulain) [reply src/GraphQl/Type/TypeBuilder.php:126]: Personally I think it's better like this indeed.

**Comment by soyuka** [comment]
This is done in  https://github.com/api-platform/core/pull/5657

**Pull request closed without merging**